test/integration/filters/wait_for_whole_request.cc

```diff
@@ -0,0 +1,51 @@
+#include <string>
+
+#include "envoy/http/filter.h"
+#include "envoy/registry/registry.h"
+#include "envoy/server/filter_config.h"
+
+#include "extensions/filters/http/common/pass_through_filter.h"
+
+#include "test/extensions/filters/http/common/empty_http_filter_config.h"
+#include "test/integration/filters/common.h"
+
+namespace Envoy {
+
+class WaitForWholeRequestStreamFilter : public Http::PassThroughFilter {
```

Comment:
nit: I think this is WaitForWholeRequestAndResponseStreamFilter or similar? (Same for other names like this.) Can you also add a small comment up top for what this does like you did with the other one?